### DISCUSSION TITLE
What is the difference between Fira Code and Fira Code Mono?

### DISCUSSION THREAD

**Discussion by Canowyrms** [ryanoasis/nerd-fonts#1039] [Q&A] [answered]
Apologies if this is a stupid question, but I haven't been able to find an answer myself.

What is the difference between 

"Fira Code Regular Nerd Font Complete Windows Compatible.ttf"

 and 

"Fira Code Regular Nerd Font Complete Mono Windows Compatible.ttf"

Should I install one over the other? Should I install both?

Thanks

**Comment by Canowyrms** [comment, marked as answer]
Naturally, I find the answer mere moments after posting the question...

Per https://github.com/ryanoasis/nerd-fonts/blob/master/patched-fonts/FiraCode/font-info.md#which-font :

> If you are limited to monospaced fonts (because of your terminal, etc) then pick a font with the 'Mono' suffix. This denotes that the Nerd Font glyphs will be monospaced not necessarily that the entire font will be monospaced.

Next question - is it okay to install both mono and non-mono? I assume it's safe.

**Reply by Finii** [reply]
It should be ok.
In the past there were some problems with the names in the fonts - they had the same name so the system could not tell them apart.
But that should be fixed (hopefully). I'm only really sure it is fixed for fonts that use the 'new naming engine' which is enabled only for 2 fonts right now, to see if it is really bugfree ;-)

**Reply by Finii** [reply]
Just checked the files currently in `master` (i.e. _not_ the release):

![image](https://user-images.githubusercontent.com/16012374/211277825-5f9e6133-4337-40b0-b009-8c1c5d47cc54.png)

The Mono and non-Mono fonts do have different names.

But you can also see that the 'old application font grouping' is broken, while the 'better application font grouping' is ok.

The font's name is a combination of Family and Subfamily. You see that all the different weights have the same Family-Subfamily, which makes them indistinguishable. The newer typographic Family-Subfamily (the rightmost 2 colums) DO have diffences, so it is fine.

Most applications and systems nowadays use the typographic Familes.

**Reply by Finii** [reply]
~~Oh no, the typographic Families are also broken, because BOLD is not labelled as such :-(~~

Ah no, it is correct. If the typographic field is empty the 'normal' field is used, an there it is "Bold".

**Reply by Canowyrms** [reply]
Okay, some of this is beyond my understanding.

I can't remember the details at this point, but I was running into some issues in some of my applications if I had multiple weights installed. Some wouldn't list all the weights correctly, therefore couldn't select the weight I wanted, things like that. If we're not meant to install more than one weight, that isn't obvious to me.

I've installed just the two non-Monospace Retina fonts (both the Windows compatible and non-Windows compatible). So far, this seems to be working the best. I can select between the two and if an application isn't rendering properly with one, I can just switch to the other.

Thank you for taking the time to look at this and reply to me.

**Reply by Finii** [reply]
> therefore couldn't select the weight I wanted

Yes, this is how 'the naming bug' shows up for users.

When a font is patched we change the name, basically add 'Nerd Font' to it. That sounds simple, and it has been implemented thus (very simple). In the beginning Nerd Font just patched single fonts, and simple was fine.
But it turns out that font renaming is not that simple, and if you have 'sets of fonts' (styles and weights), the simple code just breaks.

For that reason a very big blob of code has been added that handles the font renaming. It shall fix all the naming problems.
With the upcoming `v2.3.0` we test that with some fonts, but not all (so the other fonts will still have the known naming problems), but with `v3.0.0` and if everything goes well, this will be fixed for all fonts.

Some problems are already fixed with `v2.3.0`, for example that Mono and non-Mono _Windows Compatible_ fonts had the **same** name :sob: Fixed with a6e6ee26d8
